src/code_oscar.jl

```diff
@@ -0,0 +1,566 @@
+    sorted_orbit = sort(current_orbit; lt=leq)
+    return sorted_orbit[1]
+end;
+
+function rewrite_action_to_orbits(homs)
+    G = Oscar.domain(homs[1])
+    Ggens = Oscar.gens(G)
+    generators_new_perm = [Vector{Int}[] for x in Ggens]
+
+    for hom in homs
+      for j in 1:length(generators_new_perm)
+        img = Oscar.listperm(Oscar.image(hom, Ggens[j]))
+        if length(img) == 0
+          img = Int[1]
+        end
+        push!(generators_new_perm[j], img)
+      end
+    end
+
+    return generators_new_perm
+end;
+
+
+#############################################################################
+
+# user functions
+
+"""
+    is_monomial_free(I::Oscar.MPolyIdeal{T}, vars_to_zero = []) where {T}
+
+See Prop. 3.1 in [BKR](https://arxiv.org/abs/1603.09241).
+"""
+function is_monomial_free(I::Oscar.MPolyIdeal{T}, vars_to_zero::Vector{Int} = Int[]) where {T}
+    Oscar.singular_assure(I)
+    SingI = I.gens.S
+    R = Oscar.base_ring(SingI)
+    Rgens = Oscar.gens(R)
+    nr_variables = length(Rgens)
+    poly_list = map(f -> Oscar.evaluate(f, vars_to_zero, fill(R(0), length(vars_to_zero))), Oscar.gens(SingI))
+
+    # perm is an n-cycle
+    perm = collect(2:nr_variables)
+    push!(perm, 1)
+
+    Rgens_permuted = Rgens[perm]
+
+    for i in 1:nr_variables
+        if !(nr_variables in vars_to_zero)
+            saturated = Singular.satstd(
+                Singular.Ideal(R, poly_list), Singular.MaximalIdeal(R, 1))
+            if Singular.ngens(saturated) == 1 && saturated[1] == R(1)
+                return false
+            elseif reduce(R(1), saturated) == 0
+                return false
+            end
+        end
+        vars_to_zero = perm[vars_to_zero]
+        poly_list = [Oscar.evaluate(j, Rgens_permuted) for j in poly_list]
+    end
+
+    return true
+end
+
+
+"""
+    orbit_cones(I::Oscar.MPolyIdeal{T}, Q::Array{Int,2}, G::Oscar.GAPGroup = symmetric_group(1)) where {T}
+
+> Return orbit representatives of the group `G` on the set of those cones
+> whose defining rays are given by subsets `S` of the rows of the matrix `Q`,
+> such that the matrix `S` has full rank and such that the ideal `I` is
+> monomial-free (see [`is_monomial_free`](@ref)) w.r.t. the variables `x_i`
+> for which the `i`-th row of `Q` is not contained in `S`.
+"""
+function orbit_cones(I::Oscar.MPolyIdeal{T}, Q::Array{Int,2}, G::Oscar.GAPGroup = symmetric_group(1)) where {T}
+    nr_variables = size(Q, 1)
+    projected_dimension = size(Q, 2)
+
+    collector_cones = []
+
+    # We need not consider sets of smaller size because of the rank condition.
+    for k in projected_dimension:nr_variables
+        set = GAP.Globals.Combinations(GAP.GapObj(1:nr_variables), k)
+        orbs = GAP.Globals.Orbits(G.X, set, GAP.Globals.OnSets)
+        reps = [orbs[i][1] for i in 1:length(orbs)]
+        reps_julia = [Vector{Int64}(i) for i in reps]
+
+        for i in reps_julia
+            current_mat = Q[i,:]
```

Comment:
OK, for this you really want a sorted `Vector{Int}` and not a `Set{Int}`. (we could also work with `BitSet`, esp. if `nr_variables` is not too big, say <= 64; this would be very efficient and also cheap to convert to a `Vector{Int}`